### PR TITLE
Remove navigation bar appearance overrides

### PR DESCRIPTION
**Pull request by kean**
There are multiple problems caused by the legacy navigation bar appearance overrides.

- It [messes up](https://github.com/user-attachments/assets/913f3e5d-5686-4486-978a-4f270e554588) safe areas insets when using UIKit with SwiftUI 
- It breaks navigation bars in certain native screen, like photos picker – [#22333](https://github.com/wordpress-mobile/WordPress-iOS/issues/22333)
- It prevents the app from taking advantages of the new navigation view feature such as scroll edge appearance, and that makes the app feel outdated. The new appearance is especially beneficial for iPad.

Most of the new screens have been resetting these overrides to get the system behavior:

```swift
let viewController = UIHostingController(rootView: SubmitFeedbackView(presentingViewController: self, source: source))
viewController.configureDefaultNavigationBarAppearance()

let navigationController = UINavigationController(rootViewController: viewController)
navigationController.navigationBar.isTranslucent = true // Reset to default
```

This is not sustainable to keep doing it every screen, so it's time to bite a bullet and finally get rid of these and nearly all other global appearance overrides that break the assumptions about how the standard controls work. It will make the UI feel modern and will reduce the maintenance costs.

## Screenshots

| Before  | After |
| ------------- | ------------- |
|  profile-after | Screenshot 2024-07-25 at 10 06 09 AM |
| ipad-before | ipad-after |

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by kean** [comment]
@jkmassel wdyt?

**Comment by kean** [comment]
There are a multiple screens that need to be updated to support the standard styles. I'm also using this is as on opportunity to phase out deprecated `.grouped` table view style.

## TimeZoneSelectorViewController

More / Site Settings / Time Zone

The correct table style to use on this screen is `.plain` with sections to ensure that section headers are sticky. I also updated insets to use standard system margins. The search primary button now says "Search" instead of "done".

| Before | After |
|------|-----|
| before-1 | after-1 |
| before-2 | after-2 |

## LanguageSelectorViewController

More / Site Settings / Language

Same treatment with `.plain` style and search bar displaying using `navigationItem`.

| Before | After |
|------|-----|
|lang-before-1 |lang-after-1| 
| lang-before-2|lang-after-2 | 

## SiteTagsViewController

More / Site Settings / Tags

For this screen, I added two new SwiftUI-based APIs to replace now deprecated `DSButton` and `NoResultsViewController`.

`EmptyStateView` allows provides an easy way to create customizable empty states. Notice a new `.primary` button style designed in a "right" SwiftUI way. It also has a convenience `EmptyStateView.search()` factory method that can be used anywhere.

```swift
EmptyStateView {
    Label("You don't have any tags", systemImage: "tag")
} description: {
    Text("Tags created here can be easily added to new posts")
} actions: {
    Button {
        
    } label: {
        Text("Create a Tag")
    }
    .buttonStyle(.primary)
}
```

| Before | After |
|------|-----|
|tags-before-1 |tags-after-1 | 
|tags-before-2 | tags-after-2| 
|tags-before-3 | tags-after-3|
| tags-before-4| tags-after-4| 

## DeleteSiteViewController

| Before | After |
|------|-----|
|delete-site-before | delete-site-after|

**Comment by jkmassel** [comment]
Yep, we'll need to do this sooner or later - no objection to getting started on it now

**Comment by crazytonyli** [comment]
Awesome work! The new UI looks great.

It's pretty difficult to review the code changes, due to large amount of deleted and moved code. Let me know if there is any specific code that you think worth a closer look, otherwise I'll browse the app and see if I can spot any UI issues.

**Comment by crazytonyli** [comment]
View a notification detail, tap the site icon, the site icon in the site detail screen is misplaced.

| Before | After |
| ------ | ----- |
| IMG_5586 | IMG_5585 |

**Comment by crazytonyli** [comment]
View a notification, tap one of the tags/topics, the navigation bar push animation looks a bit strange.

https://github.com/user-attachments/assets/eea19598-09e8-457a-8c54-c6428c02b027

**Comment by crazytonyli** [comment]
Tap the Reader tab, search something, go to the Blogs tab of the search result, the tab bar becomes transparent.

IMG_5589

**Comment by kean** [comment]
> It's pretty difficult to review the code changes, due to large amount of deleted and moved code. Let me know if there is any specific code that you think worth a closer look, otherwise I'll browse the app and see if I can spot any UI issues.

I tried to make only the most basic changes possible to make it unnecessary to review it line-by-line. As long as it works, it should be good. The only code I'm a bit worried about is in `SiteStatsDashboardViewController` where I added support for swiping between tabs. In hindsight, I should've left it out of this PR.

**Comment by crazytonyli** [comment]
Found a bug in the new swiping features in stats. After dragging to the Traffic tab but returning to the original Insights tab, then tapping the Traffic tab, the page would stay on Insights tab. See the video recording below.

It's probably not obvious in the screen recording, all interactions are done by swiping, except the very last interaction to view Traffic tab which is done by tapping the tab.

https://github.com/user-attachments/assets/83b9057d-c317-4a6a-89c4-9e33aa76475b

File: WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/Jetpack.xcscheme [hunk 1]
```diff
@@ -121,11 +121,6 @@
             value = ""
             isEnabled = "YES">
          </AdditionalOption>
-         <AdditionalOption
-            key = "MallocScribble"
-            value = ""
-            isEnabled = "YES">
-         </AdditionalOption>
```
Comment (crazytonyli) [inline WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/Jetpack.xcscheme:128]: Just want to highlight this scheme change, which I think should be done: disabling Malloc Scribble diagnosis prevents WKWebView from crashing during debug sessions.

**Comment by crazytonyli** [comment]
The navigation bar on Media detail screen is fully transparent.

Screenshot 2024-07-31 at 1 38 03 PM

**Comment by kean** [comment]
> The navigation bar on Media detail screen is fully transparent.

Good catch. I don't remember testing scroll it vertically.

Fixed by reverting to the standard bar appearance using new  `WPStyleGuide.disableScrollEdgeAppearance`. It uses `UIPageViewController` and it doesn't seem to play well with scroll edge appearance. Ideally, it should be fixed differently, but I want to unblock the merge.

> Tap the Reader tab, search something, go to the Blogs tab of the search result, the tab bar becomes transparent.

Fixed, and fixed the details page. I'm not sure everything in Reader works. I'm wrapping up for today.
I'm a bit concerned about this part of the app. It needs to become much simpler to make it maintainable, but it's not an area I would invest much resources into.

> Found a bug in the new swiping features in stats.

I removed this change. As I mentioned earlier, it wasn't a good call to include it in this PR, but it will be a nice feature to add in the future.

**Comment by crazytonyli** [comment]
The tab bar shows a different style when the Reader tab is selected.

| Reader | My Site |
| ------- | ------ |
| IMG_5592 | IMG_5593 |

**Comment by crazytonyli** [comment]
The navigation bar animation during pushing search screen in Reader looks odd.

https://github.com/user-attachments/assets/46e26f85-097d-4fd7-b934-8ff0540dc370

**Comment by crazytonyli** [comment]
A modal presented from Reader has a transparent navigation bar.

Tap a reader post, scroll to the "More from ..." section at bottom of the post and tap a post, tap a tag/topic in the post in the presented modal, checkout the navigation bar in the pushed topic screen.

https://github.com/user-attachments/assets/dc660339-1496-4b17-b6e7-296ec6532846

**Comment by kean** [comment]
> The tab bar shows a different style when the Reader tab is selected.

I disabled scroll edge appearance for this tab for now, so it looks good now.

> The navigation bar animation during pushing search screen in Reader looks odd.

I fixed it and also fixed a production issue where the "recent searches" were covering the filter bar. It was apparently broken when the filter bar was added.

https://github.com/user-attachments/assets/75f2e2eb-acb9-4bd4-9798-0ab70d3d3557

> A modal presented from Reader has a transparent navigation bar.

I turned it into a push in the existing navigation stack because this is how every other navigation to the new topic/tag/site works in the Reader. I see no reason to show only related posts in a small sheet. We can't fix and maintain both types of presentation at the moment.

I also changes the background color of the stream to white/black, so there is no longer a gray gap when you use ptr.

I fixed the transition when selecting a topic in the main view as well. The issue was that the `ReaderStreamViewController` was configured to not extend below the top bars in storyboards.


@crazytonyli, I think it's an unacceptable state to get this PR merged and fix forward if there any issues. I want to make sure it's merged in trunk earlier so we don't hit big merge conflicts and the new code being developed uses the new global styles or lack thereof.

**Comment by crazytonyli** [comment]
> I think it's an unacceptable state to get this PR merged

@kean Just double checking, do you mean "acceptable"?

> [...] and fix forward if there any issues

Sure. I'll do a final walkthrough and will approve the PR if no major issue is found.

> The issue was that the ReaderStreamViewController was configured to not extend below the top bars in storyboards.

I noticed this issue on some other screens too, but didn't report them because they are small issues. But it'd be good to address when having a chance. Like the Me screen

IMG_5598

**Review by crazytonyli — APPROVED** [review]
All reported issues are addressed, and I couldn't find any major ones.

**Comment by kean** [comment]
> @kean Just double checking, do you mean "acceptable"?

Yes 🤣 

> Sure. I'll do a final walkthrough and will approve the PR if no major issue is found.

I also plan to do more testing once the site picker changes and the newer iPad changes land.

> But it'd be good to address when having a chance. Like the Me screen

I added it to my list. I have a couple of tiny other things I noticed when testing.

> All reported issues are addressed, and I couldn't find any major ones.

🕺

**Pull request merged**